private/dependency_tree_parser.bzl

```diff
@@ -319,6 +319,21 @@ genrule(
     to_return.append("alias(\n\tname = \"%s\",\n\tactual = \"%s\",\n%s)" %
                      (versioned_target_alias_label, target_label, alias_visibility))
 
+    for annotation_processor in artifact.get("annotation_processors", []):
+        to_return.append(
+            """java_plugin(
+\tname = "{name}",
+\tdeps = [":{jar_target}"],
+\tgenerates_api = True,
+processor_class = "{processor_class}",
+{alias_visibility})""".format(
+                name = "{}__java_plugin__{}".format(target_label, escape(annotation_processor)),
```

Comment:
Can you also provide a `java_plugin()` macro that hides the naming implementation detail in BUILD files?

Something like the `artifact()` macro:

https://github.com/bazelbuild/rules_jvm_external/blob/653bc80e45d317fce7492c90be0acf4c98da8990/private/rules/artifact.bzl#L4

Reply:
Nice idea! Done!